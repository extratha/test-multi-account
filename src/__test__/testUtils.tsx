import { RenderOptions, RenderResult, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SWRConfig } from 'swr';
import { theme } from '@/config/config-mui/theme';
import { ThemeProvider as MUIThemeProvider } from '@mui/material/styles';
import { NextIntlClientProvider } from 'next-intl';
import commonTh from '../../public/locales/th/common.json';
import commonEn from '../../public/locales/th/common.json';
type RenderWithProvider = (
  elm: React.ReactElement,
  renderOptions?: ProviderOptions,
) => RenderResult;

export type ProviderOptions = {
  formOptions?: Record<string, unknown>;
} & RenderOptions;

const messages = {
  ...commonTh,
  ...commonEn,
};

const renderWithProviders: RenderWithProvider = (
  component,
  renderOptions?,
  locale = 'th',
) => {
  const wrapper = ({ children }: { children: React.ReactNode }) => (
    <NextIntlClientProvider locale={locale} messages={messages}>
      <MUIThemeProvider theme={theme}>
        <SWRConfig value={{ provider: () => new Map() }}>{children}</SWRConfig>
      </MUIThemeProvider>
    </NextIntlClientProvider>
  );

  return render(component, {
    wrapper,
    ...renderOptions,
  });
};

export { renderWithProviders as render, userEvent };
export * from '@testing-library/react';
