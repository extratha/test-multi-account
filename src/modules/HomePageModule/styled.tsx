import { Stack, Typography, styled } from "@mui/material";

export const ContentContainerWrapper = styled(Stack)(({ theme }) => ({
  backgroundColor: theme.palette.background.paper,
  padding: "1rem",
  borderRadius: "20px",
  height: "100%",
  overflowY: "auto",
}));

export const ContentContainer = styled(Stack)({
  height: "100%",
  maxHeight: "100vh",
  width: "100%",
  padding: "1rem",
});

export const TypographyPageHeadline = styled(Typography)(() => ({
  width: "fit-content",
  background: "linear-gradient(90deg, #023781 0%, #028DF5 50%, #01EAFF 100%)",
  WebkitBackgroundClip: "text",
  WebkitTextFillColor: "transparent",
  backgroundClip: "text",
  textFillColor: "transparent",
}));
