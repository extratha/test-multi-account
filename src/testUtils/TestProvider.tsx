import { ThemeProvider } from "@emotion/react";

import { ReactNode } from "react";
import { SWRConfig } from "swr";

import ModalLayer from "@/components/Modal/ModalLayer";
import { theme } from "@/config/config-mui";

interface TestProviderProps {
  children?: ReactNode;
}

export const TestProvider = ({ children }: TestProviderProps) => {
  return (
    <ThemeProvider theme={theme}>
      <SWRConfig value={{ provider: () => new Map() }}>{children}</SWRConfig>
      <ModalLayer />
    </ThemeProvider>
  );
};
