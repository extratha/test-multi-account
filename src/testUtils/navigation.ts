import * as NextNavigation from "next/navigation";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
  useParams: jest.fn(),
  usePathname: jest.fn(),
  useSearchParams: jest.fn(),
  useServerInsertedHTML: jest.fn(),
}));

export interface SpyUseSearchParams {
  get: jest.Mock;
}

export interface SpyUseRouter {
  replace: jest.Mock;
  push: jest.Mock;
  back: jest.Mock;
}

export const spyUseParams = (params: any = {}) => {
  jest.spyOn(NextNavigation, "useParams").mockReturnValue(params);
  return NextNavigation.useParams as jest.Mock;
};

export const spyUseRouter = (): SpyUseRouter => {
  const replace = jest.fn();
  const push = jest.fn();
  const back = jest.fn();
  jest.spyOn(NextNavigation, "useRouter").mockReturnValue({ replace, push, back } as any);
  return { replace, push, back };
};

export const spyUsePathname = (pathname: string) => {
  jest.spyOn(NextNavigation, "usePathname").mockReturnValue(pathname);
};

export const spyUseSearchParams = (): SpyUseSearchParams => {
  const get = jest.fn();
  jest.spyOn(NextNavigation, "useSearchParams").mockReturnValue({ get } as any);
  return {
    get,
  };
};

export const spyUseServerInsertedHTML = () => {
  return jest.spyOn(NextNavigation, "useServerInsertedHTML");
};
