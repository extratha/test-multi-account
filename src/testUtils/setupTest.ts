import "@testing-library/jest-dom";
import "jest-canvas-mock";

import "@/__mocks__/Markdown";
import "@/__mocks__/MockFirebase";
import "@/__mocks__/NextImage";
import { LOCALE_LANGUAGE } from "@/config";
import { initI18next } from "@/locales/i18n";
import locale from "@/locales/th.json";

initI18next(LOCALE_LANGUAGE.TH, locale);

jest.mock("@/api/api", () => {
  const actualModule = jest.requireActual("@/api/api");
  return { ...actualModule, __esModule: true };
});

jest.mock("@/api/apiUnauthorize", () => {
  const actualModule = jest.requireActual("@/api/apiUnauthorize");
  return { ...actualModule, __esModule: true };
});

jest.mock("next/config", () => () => ({
  setConfig: jest.fn(),
  publicRuntimeConfig: {
    apiToken:
      "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE3MDY2NzUzNDQsImlhdCI6MTcwNjU4ODk0NCwicHJvZmlsZUlkIjoiZDU1YWY3NDUtOTYyZS00MTk2LTkyOTEtZDg2YTliYWNiZWY2In0.sNgwYjIGLYEVAptqILGgRVGXwt9FCPKHRubEZyaodhg",
    mockBaseApiUrl: "https://www.mock-playground-dashboard.co.th",
    baseApiUrl: "https://www.mock-playground-dashboard.co.th",
    globalEndpoint: "playground-dashboard",
    publicApiHost: "http://localhost:3000/",
  },
}));

jest.mock("cookies-next", () => ({
  __esModule: true,
  ...jest.requireActual("cookies-next"),
  getCookie: jest.fn().mockReturnValue("MOCK_ACCESS_TOKEN"),
}));
