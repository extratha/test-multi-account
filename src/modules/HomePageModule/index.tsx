'use client'
import GridMenu from "@/components/Menus/GridMenu";
import { CustomContainer } from "./styled";
import { aiMenuList, MenuItem } from "@/constant/menu";
import { Stack, Typography } from "@mui/material";
import { useTranslations } from "next-intl";
import { webPaths } from "@/constant/webPaths";

const HomePageModule = () => {
  const t = useTranslations('Common')
  return (
    <>
      <Stack height='100%' width='100%' padding='2rem'>
        <CustomContainer>
          <Stack padding='2rem' spacing={3}>
            <Typography
              variant='displayMediumSemiBold'
              sx={{
                width: 'fit-content',
                background: 'linear-gradient(90deg, #023781 0%, #028DF5 50%, #01EAFF 100%)',
                WebkitBackgroundClip: 'text',
                WebkitTextFillColor: 'transparent',
                backgroundClip: 'text',
                textFillColor: 'transparent',
              }}
            >
              {`${t('title.hello')} ${t('roles.admin')}`}
            </Typography>
            <Typography variant='displayMediumSemiBold'>{t('title.pleaseSelectMenu')} </Typography>
          </Stack>
          <GridMenu menus={aiMenuList().filter((menu: MenuItem) => menu.path !== webPaths.home)}></GridMenu>
        </CustomContainer>
      </Stack>

    </>
  )
}
export default HomePageModule;