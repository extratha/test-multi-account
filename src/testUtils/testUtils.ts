import { act, render as testRender } from "@testing-library/react";
import dayjs from "dayjs";
import flushPromises from "flush-promises";
import { ReactElement } from "react";

import { TestProvider } from "./TestProvider";

export const API = {
  LOGIN: "/auth/login",
  EXAMPLE_AI_INTERPRET_URL: "/lab/examples/labId",
  GET_LAB_INTERPRETS_ID_URL: "/lab/interprets/labId",
  SUBMIT_LAB_INTERPRETS_URL: "/lab/interprets",
  CONSENT: "/consents",
  TERMS_AND_CONDITIONS: "/consents/terms-conditions",
  TERMS_AND_CONDITIONS_LATEST: "/consents/terms-conditions/latest",
  PRIVACY_POLICY: "/consents/privacy-policies",
  PRIVACY_POLICY_LATEST: "/consents/privacy-policies/latest",
  PUBLIC_TERMS_CONDITIONS: "/consents/terms-conditions/public",
  PUBLIC_PRIVACY_POLICIES: "/consents/privacy-policies/public",
};

export const render = (ui: ReactElement) => {
  return testRender(ui, { wrapper: TestProvider });
};

export const flushPromise = async () => {
  await act(async () => {
    await flushPromises();
  });
};

export const advanceTimersByTime = async (time: number) => {
  await act(async () => {
    jest.advanceTimersByTime(time);
  });
};

export const setMockDate = (date: string) => {
  jest.setSystemTime(dayjs(date).valueOf());
};
