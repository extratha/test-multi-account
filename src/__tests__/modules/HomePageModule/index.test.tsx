import HomePageModule from "@/modules/HomePageModule";
import { render } from "@/testUtils/testUtils";

jest.mock("next/navigation", () => ({
  ...jest.requireActual("next/navigation"),
  useRouter: jest.fn(),
}));

describe("HomePageModule", () => {
  it("should render correctly", async () => {
    const { asFragment } = render(<HomePageModule />);
    expect(asFragment()).toMatchSnapshot();
  });
});
