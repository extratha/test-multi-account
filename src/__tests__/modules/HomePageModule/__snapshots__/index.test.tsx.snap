// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageModule to match snap 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-11ifw6v-MuiStack-root"
  >
    <div
      class="MuiStack-root css-16ptvjd-MuiStack-root"
    >
      <div
        class="MuiStack-root css-1f3kc6o-MuiStack-root"
      >
        <h1
          class="MuiTypography-root MuiTypography-displayMediumSemiBold css-16c1pey-MuiTypography-root"
        >
          สวัสดี แอดมิน
        </h1>
        <h1
          class="MuiTypography-root MuiTypography-displayMediumSemiBold css-1p3qgsj-MuiTypography-root"
        >
          โปรดเลือกเครื่องมือที่คุณต้องการ 
        </h1>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-1vp1qdv-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-au940r-MuiGrid-root"
        >
          <div
            class="MuiStack-root css-kk7l7a-MuiStack-root"
          >
            <img
              alt=""
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent; width: 48px; height: 48px; margin: 0px 0px 8px 0px;"
              width="40"
            />
            <h3
              class="MuiTypography-root MuiTypography-titleLargeSemiBold css-1puhohr-MuiTypography-root"
            >
              AI Interpret
            </h3>
            <span
              class="MuiTypography-root MuiTypography-bodyLarge css-uzzo47-MuiTypography-root"
            >
              แปลและสรุปผลสุขภาพ
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
